certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -867,3 +875,18 @@ def install_ssl_options_conf(options_ssl):
     # Check to make sure options-ssl.conf is installed
     if not os.path.isfile(options_ssl):
         shutil.copyfile(constants.MOD_SSL_CONF_SRC, options_ssl)
+        return
+    # there's already a file there. if it exactly matches a previous file hash,
+    # we can update it. otherwise, create a .new and print a warning.
+    digest = crypto_util.sha256sum(options_ssl)
+    if digest in PREVIOUS_SSL_OPTIONS_HASHES: # safe to update
+        shutil.copyfile(constants.MOD_SSL_CONF_SRC, options_ssl)
+    elif digest == CURRENT_SSL_OPTIONS_HASH: # already up to date
+        return
+    else: # not safe to update
+        new_filename = options_ssl + ".new"
+        shutil.copyfile(constants.MOD_SSL_CONF_SRC, new_filename)
+        # print warning
+        logger.warning("%s has been manually modified; updated ssl configuration options "
```

Comment:
We could also do checking of certbot version -- only do things once per version. I hasn't wanted to do that because it can get messy, but if we don't always want to create `.new`, there's various ways we could work version number in.